js/src/api/contract/contract.js

```diff
@@ -269,6 +217,11 @@ export default class Contract {
       }
     }
 
+    return this._subscribe(event, options, callback);
+  }
+
+  _subscribe (event = null, _options, callback) {
+    const subscriptionId = this._subscriptions.length;
```

Comment:
Or maybe just use the filterId throughout instead of mapping(subscriptionId => filterId). Iteration can go through everything, and unsubscribe uses delete on the object (to clear subscriptions), publishing iterates through the values.